src/main/kotlin/com/github/mitchellolsthoorn/testgenie/toolwindow/TestGenieToolWindow.kt

```diff
@@ -2,75 +2,194 @@ package com.github.mitchellolsthoorn.testgenie.toolwindow
 
 /**
  * This class stores the UI of the TestGenie tool window.
  */
 class TestGenieToolWindow(_toolWindow: ToolWindow) {
 
-    private val toolWindow = _toolWindow
+    private val toolWindow : ToolWindow = _toolWindow
 
-    private var maxSizeTextField: JTextField? = null
-    private var globalTimeOutTextField: JTextField? = null
-    private var coverageCombobox: JComboBox<Boolean>? = null
-    private var saveButton: JButton? = null
+    private var searchBudget: JSpinner = JSpinner(SpinnerNumberModel(60,0,10000,1))
+    private var localSearchBudgetType: ComboBox<LocalSearchBudgetType> = ComboBox(LocalSearchBudgetType.values())
+    private var localSearchBudgetValue: JSpinner = JSpinner(SpinnerNumberModel(5,0,10000,1))
+    private var stoppingCondition: ComboBox<StoppingCondition> = ComboBox<StoppingCondition>(StoppingCondition.values())
+    private var initializationTimeout: JSpinner = JSpinner(SpinnerNumberModel(120,0,10000,1))
+    private var minimisationTimeout: JSpinner = JSpinner(SpinnerNumberModel(60,0,10000,1))
+    private var assertionTimeout: JSpinner = JSpinner(SpinnerNumberModel(60,0,10000,1))
+    private var junitCheckTimeout: JSpinner = JSpinner(SpinnerNumberModel(60,0,10000,1))
+    private var population: JSpinner = JSpinner(SpinnerNumberModel(50,0,10000,1))
+    private var populationLimit: ComboBox<PopulationLimit> = ComboBox(PopulationLimit.values())
 
-    private var toolWindowPanel: JPanel? = null
+    private val panelTitle = JLabel("Frequently Used Parameters")
+    private var saveButton: JButton = JButton("Save")
+    private var resetButton: JButton = JButton("Reset")
+
+    private var toolWindowPanel: JPanel = JPanel()
 
     init {
-        saveButton?.addActionListener { addListenerForSaveButton(it) }
+        loadState()
+
+        toolWindowPanel = createToolWindowPanel()
+
+        panelTitle.font = Font("Monochrome", Font.BOLD, 20)
+        resetButton.toolTipText = "Reset all parameters to their default values"
+
+        saveButton.addActionListener { addListenerForSaveButton(it) }
+        resetButton.addActionListener { addListenerForResetButton(it) }
+    }
+
+    /**
+     * Creates the entire tool window panel.
+     */
+    private fun createToolWindowPanel() = FormBuilder.createFormBuilder()
+            .setFormLeftIndent(30)
+            .addVerticalGap(25)
+            .addComponent(panelTitle)
+            .addLabeledComponent(customLabel("Search budget", "Maximum search duration."), searchBudget, 25, false)
+            .addTooltip("Default: 60 seconds")
+            .addLabeledComponent(customLabel("Local search budget type", "Interpretation of local search budget value."), localSearchBudgetType, 20, false)
+            .addTooltip("Default: Time")
+            .addLabeledComponent(customLabel("Local search budget value", "Maximum budget usable for improving individuals per local search."), localSearchBudgetValue, 20, false)
+            .addTooltip("Default: 5")
```

Comment:
What do you think about having the tooltips underneath the main text instead of under the components?

Reply:
I do not have a preference for this. Not sure if we should change this now, maybe we can present it to the client and see if they have a preference.